src/dune_rules/dune_package.ml

```diff
@@ -92,7 +92,7 @@ module Lib = struct
        ; field_o "main_module_name" Module_name.encode main_module_name
        ; field_l "modes" sexp (Lib_mode.Map.Set.encode modes)
        ; field_l "obj_dir" sexp (Obj_dir.encode obj_dir)
-       ; field_o "modules" Modules.encode modules
+       ; field_o "modules" (Modules.encode ~src_dir:package_root) modules
```

Comment:
I thought we'd rather want to start from the library where the dune file is defined for the `(library ...)` rather than the root of the project. What's the rationale for it being the package_root instead?

- in `node_modules`, public libraries would get the same layout: `node_modules/LIB_NAME/<sources>`
- for private libraries, we could also add them under node_modules by adding `bs-package-name` and solve the comment added in https://github.com/ocaml/dune/pull/6841

Reply:
Just to be consistent with how we encode paths in the rest of the dune-package file. It actually doesn't matter, we decode the into the full path of the installed source anyway. So no matter how we encode it, we'll get the same path back.

Reply:
We could have even used absolute paths when encoding the dune-package files. The only reason we don't is to make the installed artifacts somewhat relocatable.